File: e-nutrition-symfony/src/Entity/Composition.php
<?php

namespace App\Entity;

use App\Repository\CompositionRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups as Groups;

/**
 * @ORM\Entity(repositoryClass=CompositionRepository::class)
 */
class Composition
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="float")
     * @Groups ("plat:read")
     */
    private $poid;

    /**
     * @ORM\ManyToOne(targetEntity=Aliment::class, inversedBy="compositions" , cascade={"persist"})
     * @Groups ("plat:read")
     */
    private $aliment;

    /**
     * @ORM\ManyToOne(targetEntity=Plat::class, inversedBy="compostions")
     */
    private $plat;


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPoid(): ?float
    {
        return $this->poid;
    }

    public function setPoid(float $poid): self
    {
        $this->poid = $poid;

        return $this;
    }

    public function getAliment(): ?Aliment
    {
        return $this->aliment;
    }

    public function setAliment(?Aliment $aliment): self
    {
        $this->aliment = $aliment;

        return $this;
    }

    public function getPlat(): ?Plat
    {
        return $this->plat;
    }

    public function setPlat(?Plat $plat): self
    {
        $this->plat = $plat;

        return $this;
    }
    public function calculerCalorieParpoid() :float
    {
        return (($this->getAliment()->calculerCalorie()/$this->getAliment()->getPoid())*$this->poid);
    }



}

File: e-nutrition-java/src/e/nutrition/Models/Composition.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package e.nutrition.Models;

/**
 *
 * @author bacem
 */
public class Composition {
   private int id;
   private float poid;
   private Aliment aliment;
   private Plat plat;

    @Override
    public String toString() {
        return "Composition{" + "id=" + id + ", poid=" + poid + ", aliment=" + aliment + ", plat=" + plat + '}';
    }
   
   public float calculerCalorieParpoid()
   {
   return ( (this.aliment.getCalculerCalorie()/this.aliment.getPoid())*this.poid )   ; 
   }

    public Composition(int id, float poid, Aliment aliment, Plat plat) {
        this.id = id;
        this.poid = poid;
        this.aliment = aliment;
        this.plat = plat;
    }

    public Composition(float poid, Aliment aliment, Plat plat) {
        this.poid = poid;
        this.aliment = aliment;
        this.plat = plat;
    }

    public float getPoid() {
        return poid;
    }

    public void setPoid(float poid) {
        this.poid = poid;
    }

    public Aliment getAliment() {
        return aliment;
    }

    public void setAliment(Aliment aliment) {
        this.aliment = aliment;
    }

    public Plat getPlat() {
        return plat;
    }

    public void setPlat(Plat plat) {
        this.plat = plat;
    }

    public int getId() {
        return id;
    }
    
   
   
}
